src/main/java/com/google/javascript/gents/CommentLinkingPass.java

```diff
@@ -223,12 +241,14 @@ public final boolean shouldTraverse(NodeTraversal nodeTraversal, Node n, Node pa
       } else if (getLastLineOfCurrentComment() == line - 1) {
         // Comment ends just before code
         linkCommentBufferToNode(n);
-      } else if (!hasRemainingComments()) {
-        // Exhausted all comments, output floating comment before current node
-        parent.addChildBefore(newFloatingCommentFromBuffer(), n);
+      } else if (!hasRemainingComments() && !outputCommentAfterLine) {
+        // Exhausted all comments, output floating comment before current node or at parent node.
```

Comment:
We haven't necessarily exhausted all comments yet. It's possible that we just outputted the current group of comments in the while loop above and shifted the next comment in buffer (which is the real last comment, therefore `!hasRemaingComments()`).